### DISCUSSION TITLE
How can I get a list of all components from within a module?

### DISCUSSION THREAD

**Discussion by sz55net** [nuxt/nuxt#15960] [Questions] [answered]
I need to be able to access the list and their "name" after build, so simply iterating through the components directory won't work.

**Comment by jskitz** [comment, marked as answer]
Are you using Nuxt 3? If so, there is a file called `components.d.ts` that will list all components that can be auto imported into your project. This file is located in the directory `.nuxt` which is a hidden directory. If you are using Nuxt 2, I'm not sure how that is structured if you turn on auto importing.
